pymc3/distributions/continuous.py

```diff
@@ -1113,3 +1113,43 @@ def logp(self, value):
                       - T.log(sigma * T.sqrt(2 * np.pi))
                       - 0.5 * ((value - mu) / sigma)**2)
         return bound(lp, sigma > 0., nu > 0.)
+
+
+class VonMises(Continuous):
+    R"""
+    Univariate VonMises log-likelihood.
+    .. math::
+        f(x \mid \mu, \kappa) =  
+            \frac{e^{\kappa\cos(x-\mu)}}{2\pi I_0(\kappa)}
+             
+    where :I_0 is the modified Bessel function of order 0.
+    
+    ========  ==========================================
+    Support   :math:`x \in [-\pi, \pi]`
+    Mean      :math:`\mu`
+    Variance  :math:`\frac{1-I1(\kappa)}{I0(\kappa)}`
+    ========  ==========================================
+    Parameters
+    ----------
+    mu : float
+        Mean.
+    kappa : float
+        Concentration (\frac{1}{kappa} is analogous to \sigma^2).
+    """
+    def __init__(self, mu=0.0, kappa=None, *args, **kwargs):
+        super(VonMises, self).__init__(*args, **kwargs)
+        self.mean = self.median = self.mode = self.mu = mu
+        self.kappa = kappa 
+        self.variance = 1. - special.i1(self.kappa) / special.i0(self.kappa)
```

Comment:
I suppose it makes sense since wouldn't work because variance will be a theano expression so you can't mix in scipy / python code. Is there a `i1` and `i0` in theano?